libr/core/linux_heap_glibc.c

```diff
@@ -849,56 +853,114 @@ void GH(print_heap_fastbin)(RCore *core, GHT m_arena, MallocState *main_arena, G
 	}
 }
 
-static void GH(print_tcache_instance)(RCore *core, GHT m_arena, MallocState *main_arena) {
-	if (!core || !core->dbg || !core->dbg->maps) {
-		return;
+static void GH(parse_tcache)(RCore *core, GH(RHeapTcache) *tcache_heap) {
+	GHT tcache_fd = GHT_MAX;
+	GHT tcache_tmp = GHT_MAX;
+	RConsPrintablePalette *pal = &r_cons_singleton ()->context->pal;
+	size_t i;
+	for (i = 0; i < TCACHE_MAX_BINS; i++) {
+		if (tcache_heap->counts[i] > 0) {
+			PRINT_GA ("bin :");
+			PRINTF_BA ("%2d",i);
+			PRINT_GA (", items :");
+			PRINTF_BA ("%2d",tcache_heap->counts[i]);
+			PRINT_GA (", fd :");
+
+			PRINTF_BA ("0x%"PFMT64x, (GHT)(tcache_heap->entries[i]) - GH(HDR_SZ));
+			if (tcache_heap->counts[i] > 1) {
+				tcache_fd = (GHT) tcache_heap->entries[i];
+				size_t n;
+				for (n = 1; n < tcache_heap->counts[i]; n++) {
+					bool r = r_io_read_at (core->io, tcache_fd, (ut8 *)&tcache_tmp, sizeof (GHT));
+					if (r) {
+						tcache_tmp = r_read_le (&tcache_tmp);
+						PRINTF_BA ("->0x%"PFMT64x, tcache_tmp - TC_HDR_SZ);
+						tcache_fd = tcache_tmp;
+					} else {
+						break;
+					}
+				}
+			}
+			PRINT_BA ("\n");
+		}
+	}
+}
+
+static void GH(parse_tcache_old)(RCore *core, GH(RHeapTcacheOld) *tcache_heap) {
```

Comment:
this is a lot of duplicated code. We should find a way to write it just one time, otherwise it will be easy in the future to partially modify one code but not the other. Maybe use a macro?